distributed/tests/test_variable.py

```diff
@@ -39,6 +39,12 @@ def test_variable(c, s, a, b):
         assert time() < start + 5
 
 
+def test_delete_unset_variable(client):
+    x = Variable()
+    assert x.client is client
+    x.delete()
```

Comment:
I encourage you to use `gen_cluster` whenever possible when testing.  This is for a few reasons:

1.  If we need to debug we can enter a breakpoint anywhere.  `gen_cluster` sets up everything to run in the main thread
2. It's much faster than the `client` fixture, which while more comfortable because it uses the synchronous syntax, also sets up some processes
3.  It will force you to learn some async things if you don't know them already

https://distributed.dask.org/en/latest/develop.html#writing-tests

Reply:
I modified the test to use `gen_cluster` and `captured_logger`, thanks for the suggestion.  

I needed to trigger an action on the client after the delete to ensure the error was logged.  I used `c.close` for this, was curious is there is a better way to do this.